tags.go

```diff
@@ -0,0 +1,219 @@
+package godo
+
+import "fmt"
+
+const tagsBasePath = "v2/tags"
+
+// TagsService is an interface for interfacing with the tags
+// endpoints of the DigitalOcean API
+// See: https://developers.digitalocean.com/documentation/v2#tags
+type TagsService interface {
+	List(*ListOptions) ([]Tag, *Response, error)
+	Get(string) (*Tag, *Response, error)
+	Create(*TagCreateRequest) (*Tag, *Response, error)
+	Update(string, *TagUpdateRequest) (*Response, error)
+	Delete(string) (*Response, error)
+
+	TagResources(string, *TagResourcesRequest) (*Response, error)
+	UntagResources(string, *UntagResourcesRequest) (*Response, error)
+}
+
+// TagsServiceOp handles communication with tag related method of the
+// DigitalOcean API.
+type TagsServiceOp struct {
+	client *Client
+}
+
+var _ TagsService = &TagsServiceOp{}
+
+// Resource represent a single resource for associating/dissociating with tags
+type Resource struct {
+	ID   string `json:"resource_id,omit_empty"`
+	Type string `json:"resource_type,omit_empty"`
```

Comment:
`Type` is a known value (only droplet for now), You could create a go style enum and make this type `ResourceType`, so you can ensure this will always be a known value.

```go
type ResourceType string

const {
  DropletResourceType ResourceType = "droplet"
}
```